esp/GBPlay/main/hardware/wifi.c

```diff
@@ -58,7 +58,7 @@ static void _on_connect(void* arg, esp_event_base_t event_base, int32_t event_id
 
 void wifi_initialize()
 {
-    s_wifi_lock = xSemaphoreCreateRecursiveMutex();
+    s_wifi_lock = xSemaphoreCreateMutex();
```

Comment:
No longer needs to be recursive because `wifi_connect()` now calls the internal `_disconnect()` function which does not grab the lock.